frontend/src/core/comments/components/CommentsList.js

```diff
@@ -42,17 +42,35 @@ export default function CommentsList(props: Props) {
 
     const translationId = translation ? translation.pk : null;
 
+    const renderComment = (comment) => {
+        return (
+            <Comment
+                comment={comment}
+                canPin={canPin}
+                key={comment.id}
+                togglePinnedStatus={togglePinnedStatus}
+            />
+        );
+    };
+
+    const [pinnedComments, unpinnedComments] = comments.reduce(
+        (comment, currentComment) => {
+            comment[currentComment.pinned === true ? 0 : 1].push(
+                currentComment,
+            );
+            return comment;
+        },
+        [[], []],
+    );
+
     return (
         <div className='comments-list'>
-            <ul>
-                {comments.map((comment) => (
-                    <Comment
-                        comment={comment}
-                        canPin={canPin}
-                        key={comment.id}
-                        togglePinnedStatus={togglePinnedStatus}
-                    />
-                ))}
+            <ul className='pinned-comments'>
+                {pinnedComments.map((comment) => renderComment(comment))}
+            </ul>
+
+            <ul className='unpinned-comments'>
+                {unpinnedComments.map((comment) => renderComment(comment))}
```

Comment:
For better readability in the chronologic order, I'd repeat pinned comments below. So it should be pinned comments and then all comments, instead of pinned and unpinned.

Reply:
This isn't resolved - we still iterate over `unpinnedComments` comments, instead of (all) `comments`.